README.md
[![Flutter Community: flutter_whatsnew](https://fluttercommunity.dev/_github/header/flutter_whatsnew)](https://github.com/fluttercommunity/community)

[![Buy Me A Coffee](https://img.shields.io/badge/Donate-Buy%20Me%20A%20Coffee-yellow.svg)](https://www.buymeacoffee.com/rodydavis)
[![Donate](https://img.shields.io/badge/Donate-PayPal-green.svg)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=WSH3GVC49GNNJ)

# flutter_whatsnew

[![pub package](https://img.shields.io/pub/v/flutter_whatsnew.svg)](https://pub.dartlang.org/packages/flutter_whatsnew)

A Flutter Plugin to Show a Whats New page. 

Online Demo: https://fluttercommunity.github.io/flutter_whatsnew/

## Usage

To use this plugin, add `flutter_whatsnew` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).

``` dart
// Import package
import 'package:flutter_whatsnew/flutter_whatsnew.dart';
```

![changelog](https://github.com/fluttercommunity/flutter_whatsnew/blob/master/doc/screenshots/changes.png)

If using `WhatsNewPage.changelog` and you want it to pickup the `CHANGELOG.md' add it to the Flutter asset directory, otherwise manually pass in the .md file with the changes. This uses the full markdown previewer.

``` yaml
  assets:
    - CHANGELOG.md
```

![changelog](https://github.com/fluttercommunity/flutter_whatsnew/blob/master/doc/screenshots/changelog.png)

## Example

``` dart
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter_whatsnew/flutter_whatsnew.dart';

void main() {
  runApp(MaterialApp(home: ShowWhatsNew()));
}

class ShowWhatsNew extends StatelessWidget {
  final double textScaleFactor = 1.0;

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      body: SafeArea(
          child: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: <Widget>[
            RaisedButton(
              child: Text("Show Changelog"),
              onPressed: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(
                    builder: (context) => WhatsNewPage.changelog(
                          title: Text(
                            "What's New",
                            textScaleFactor: textScaleFactor,
                            textAlign: TextAlign.center,
                            style: const TextStyle(
                              // Text Style Needed to Look like iOS 11
                              fontSize: 22.0,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                          buttonText: Text(
                            'Continue',
                            textScaleFactor: textScaleFactor,
                            style: const TextStyle(color: Colors.white),
                          ),
                        ),
                    fullscreenDialog: true,
                  ),
                );
              },
            ),
            Container(height: 50.0),
            RaisedButton(
              child: Text("Show Changes"),
              onPressed: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(
                    builder: (context) => WhatsNewPage(
                          title: Text(
                            "What's New",
                            textScaleFactor: textScaleFactor,
                            textAlign: TextAlign.center,
                            style: const TextStyle(
                              // Text Style Needed to Look like iOS 11
                              fontSize: 22.0,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                          buttonText: Text(
                            'Continue',
                            textScaleFactor: textScaleFactor,
                            style: const TextStyle(color: Colors.white),
                          ),
                          // Create a List of WhatsNewItem for use in the Whats New Page
                          // Create as many as you need, it will be scrollable
                          items: <ListTile>[
                            ListTile(
                              leading: const Icon(Icons.color_lens),
                              title: Text(
                                'Dark Theme',
                                textScaleFactor: textScaleFactor,
                              ), //Title is the only Required Item
                              subtitle: Text(
                                'Black and grey theme (Tap to Change)',
                                textScaleFactor: textScaleFactor,
                              ),
                              onTap: () {
                                // You Can Navigate to Locations in the App
                                Navigator.of(context).pushNamed("/settings");
                              },
                            ),
                            ListTile(
                              leading: const Icon(Icons.map),
                              title: Text(
                                'Google Maps',
                                textScaleFactor: textScaleFactor,
                              ),
                              subtitle: Text(
                                'Open Address Links in Google Maps instead of Apple Maps (Tap to Change)',
                                textScaleFactor: textScaleFactor,
                              ),
                              onTap: () {
                                // You Can Navigate to Locations in the App
                                Navigator.of(context).pushNamed("/settings");
                              },
                            ),
                            ListTile(
                              leading: const Icon(Icons.notifications_active),
                              title: Text(
                                'Push Notifications',
                                textScaleFactor: textScaleFactor,
                              ),
                              subtitle: Text(
                                'Stay tuned for important information that can be pushed to you',
                                textScaleFactor: textScaleFactor,
                              ),
                              onTap: () {
                                WhatsNewPage.showDetailPopUp(
                                  context,
                                  'Info',
                                  "You can turn off push notifications any time in your application settings.",
                                );
                              },
                            ),
                          ], //Required
                        ),
                    fullscreenDialog: true,
                  ),
                );
              },
            ),
          ],
        ),
      )),
    );
  }
}

```


CHANGELOG.md
## 1.0.3

* Adding Web Support

## 1.0.2

* Adding Path Option
* Removing Native Widgets

## 1.0.1

* Bumping Version

## 1.0.0

* Addign Cupertino UI

## 0.3.0 - 04.06.2019

* Making Example Desktop Aware

## 0.2.1 - 04.05.2019

* `tb2761` Added Button Colors

## [0.2.0] - Breaking Changes

* Updated Example
* Added Ability to Read from Changelog
* Removed Method WhatsNewPage.show()
* Adding `flutter_markdown`

## [0.1.0] - Breaking Change

* Updated Example
* Removed `get_version`
* Plugin is now pure dart
* Added Method WhatsNewPage.show()

## [0.0.7] - New Button Action

* Updated Example

## [0.0.6] - New Button Action

* Added Optional Callback for Button Press.
* By Defualt it will Push to the Passed Widget.
* Updating Dart Dependency

## [0.0.5] - Updating for Flutter

* Increased button width on android.
* Updated native_widgets 0.1.0

## [0.0.4] - Bug Fixes

* Fixed Loading Bug.

## [0.0.3] - Bug Fixes

* Updating Startup Behavior.

## [0.0.2] - Bug Fixes

* Updating Startup Behavior.

## [0.0.1] - Whats New Widget

* Added Components, Version Check and Example.
* Added `native_widgets`


test/flutter_whatsnew_test.dart
import 'package:test/test.dart';

import 'package:flutter_whatsnew/flutter_whatsnew.dart';
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';

void main() {
  final double textScaleFactor = 1.0;

  test('Check for Null', () {
    final widget = WhatsNewPage(
      title: Text(
        "What's New",
        textScaleFactor: textScaleFactor,
        textAlign: TextAlign.center,
        style: TextStyle(
          // Text Style Needed to Look like iOS 11
          fontSize: 22.0,
          fontWeight: FontWeight.bold,
        ),
      ),
      buttonText: Text(
        'Continue',
        textScaleFactor: textScaleFactor,
        style: TextStyle(
          color: Colors.white,
        ),
      ),
      // Create a List of WhatsNewItem for use in the Whats New Page
      // Create as many as you need, it will be scrollable
      items: <ListTile>[
        ListTile(
          leading: Icon(Icons.color_lens),
          title: Text(
            'Dark Theme',
            textScaleFactor: textScaleFactor,
          ), //Title is the only Required Item
          subtitle: Text(
            'Black and grey theme (Tap to Change)',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // You Can Navigate to Locations in the App
            // Navigator.of(context).pushNamed("/settings");
          },
        ),
        ListTile(
          leading: Icon(Icons.map),
          title: Text(
            'Google Maps',
            textScaleFactor: textScaleFactor,
          ),
          subtitle: Text(
            'Open Address Links in Google Maps instead of Apple Maps (Tap to Change)',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // You Can Navigate to Locations in the App
            // Navigator.of(context).pushNamed("/settings");
          },
        ),
        ListTile(
          leading: Icon(Icons.notifications_active),
          title: Text(
            'Push Notifications',
            textScaleFactor: textScaleFactor,
          ),
          subtitle: Text(
            'Stay tuned for important information that can be pushed to you',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // WhatsNewPage.showDetailPopUp(
            //   context,
            //   'Info',
            //   "You can turn off push notifications any time in your application settings.",
            // );
          },
        ),
        ListTile(
          leading: Icon(Icons.arrow_drop_down),
          title: Text(
            'Dropdown Actions',
            textScaleFactor: textScaleFactor,
          ),
          subtitle: Text(
            'New ability to call cell, home and office number from the list dropdown',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // You Can Navigate to Locations in the App
            // WhatsNewPage.showDetailPopUp(
            //   context,
            //   'Info',
            //   "Tap on the down arrow on any list item to display phone numbers and emails.",
            // );
          },
        ),
        ListTile(
          leading: Icon(Icons.info),
          title: Text(
            'Dashboard Alert Icons',
            textScaleFactor: textScaleFactor,
          ),
          subtitle: Text(
            'Depending on the alert type, the icon and color for the list item will change',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // You Can Navigate to Locations in the App
            // Navigator.of(context).pushNamed("/menu");
          },
        ),
        ListTile(
          leading: Icon(Icons.notifications_none),
          title: Text(
            'Loan Milestone Icons',
            textScaleFactor: textScaleFactor,
          ),
          subtitle: Text(
            'Depending on the Loan Milestone, the icon and color for the list item will change',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // You Can Navigate to Locations in the App
            // Navigator.of(context).pushNamed("/menu");
          },
        ),
        ListTile(
          leading: Icon(Icons.person_outline),
          title: Text(
            'Loan Contacts Enhancements',
            textScaleFactor: textScaleFactor,
          ),
          subtitle: Text(
            'Updated look for faster navigation',
            textScaleFactor: textScaleFactor,
          ),
          onTap: () {
            // WhatsNewPage.showDetailPopUp(
            //   context,
            //   'Info',
            //   "Navigate to any loan then select the bottom right icon to go to the contacts. You can press the dropdown arrow for contact information.",
            // );
          },
        ),
      ], //Required
      home: HomePage(),
      showNow: false,
      showOnVersionChange: true,
    );
    expect(widget.items, !null);
  });
}

// Stateful widget for managing name data
class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

// State for managing fetching name data over HTTPP>
class _HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        // backgroundColor: globals.isDarkTheme ? null : Colors.white,
        title: Text(
          "Home",
          // textScaleFactor: globals.textScaleFactor,
        ),
      ),
      body: SingleChildScrollView(
          child: SafeArea(
        child: ListBody(
          children: <Widget>[
            Text(
              'This is a Test',
            ),
          ],
        ),
      )),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter_whatsnew/flutter_whatsnew.dart';

import 'dart:io' show Platform;
import 'package:flutter/foundation.dart';

// The existing imports
// !! Keep your existing impots here !!

/// main is entry point of Flutter application
void main() {
  // Desktop platforms aren't a valid platform.
  if (!kIsWeb) _setTargetPlatformForDesktop();
  return runApp(MyApp());
}

/// If the current platform is desktop, override the default platform to
/// a supported platform (iOS for macOS, Android for Linux and Windows).
/// Otherwise, do nothing.
void _setTargetPlatformForDesktop() {
  TargetPlatform targetPlatform;
  if (Platform.isMacOS) {
    targetPlatform = TargetPlatform.iOS;
  } else if (Platform.isLinux || Platform.isWindows) {
    targetPlatform = TargetPlatform.android;
  }
  if (targetPlatform != null) {
    debugDefaultTargetPlatformOverride = targetPlatform;
  }
}

class MyApp extends StatelessWidget {
  final double textScaleFactor = 1.0;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: HomeScreen(textScaleFactor: textScaleFactor),
    );
  }
}

class HomeScreen extends StatelessWidget {
  const HomeScreen({
    Key key,
    @required this.textScaleFactor,
  }) : super(key: key);

  final double textScaleFactor;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: <Widget>[
              RaisedButton(
                child: Text("Show Changelog"),
                onPressed: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(
                      builder: (context) => WhatsNewPage.changelog(
                        title: Text(
                          "What's New",
                          textScaleFactor: textScaleFactor,
                          textAlign: TextAlign.center,
                          style: const TextStyle(
                            // Text Style Needed to Look like iOS 11
                            fontSize: 22.0,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                        buttonText: Text(
                          'Continue',
                          textScaleFactor: textScaleFactor,
                          style: const TextStyle(color: Colors.white),
                        ),
                      ),
                      fullscreenDialog: true,
                    ),
                  );
                },
              ),
              Container(height: 50.0),
              RaisedButton(
                child: Text("Show Changes"),
                onPressed: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(
                      builder: (context) => WhatsNewPage(
                        title: Text(
                          "What's New",
                          textScaleFactor: textScaleFactor,
                          textAlign: TextAlign.center,
                          style: const TextStyle(
                            // Text Style Needed to Look like iOS 11
                            fontSize: 22.0,
                            fontWeight: FontWeight.bold,
                          ),
                        ),
                        buttonText: Text(
                          'Continue',
                          textScaleFactor: textScaleFactor,
                          style: const TextStyle(color: Colors.white),
                        ),
                        // Create a List of WhatsNewItem for use in the Whats New Page
                        // Create as many as you need, it will be scrollable
                        items: <ListTile>[
                          ListTile(
                            leading: const Icon(Icons.color_lens),
                            title: Text(
                              'Dark Theme',
                              textScaleFactor: textScaleFactor,
                            ), //Title is the only Required Item
                            subtitle: Text(
                              'Black and grey theme (Tap to Change)',
                              textScaleFactor: textScaleFactor,
                            ),
                            onTap: () {
                              // You Can Navigate to Locations in the App
                              Navigator.of(context).pushNamed("/settings");
                            },
                          ),
                          ListTile(
                            leading: const Icon(Icons.map),
                            title: Text(
                              'Google Maps',
                              textScaleFactor: textScaleFactor,
                            ),
                            subtitle: Text(
                              'Open Address Links in Google Maps instead of Apple Maps (Tap to Change)',
                              textScaleFactor: textScaleFactor,
                            ),
                            onTap: () {
                              // You Can Navigate to Locations in the App
                              Navigator.of(context).pushNamed("/settings");
                            },
                          ),
                          ListTile(
                            leading: const Icon(Icons.notifications_active),
                            title: Text(
                              'Push Notifications',
                              textScaleFactor: textScaleFactor,
                            ),
                            subtitle: Text(
                              'Stay tuned for important information that can be pushed to you',
                              textScaleFactor: textScaleFactor,
                            ),
                            onTap: () {
                              WhatsNewPage.showDetailPopUp(
                                context,
                                'Info',
                                "You can turn off push notifications any time in your application settings.",
                              );
                            },
                          ),
                        ], //Required
                      ),
                      fullscreenDialog: true,
                    ),
                  );
                },
              ),
            ],
          ),
        ),
      ),
    );
  }
}


flutter_service_worker.js
'use strict';
const CACHE_NAME = 'flutter-app-cache';
const RESOURCES = {
  "/index.html": "e36211d86e9c70409c58295974f749a1",
"/manifest.json": "00e0b69b49487ce4f9ff0c5fac8fda49",
"/icons/Icon-512.png": "96e752610906ba2a93c65f8abe1645f1",
"/icons/Icon-192.png": "ac9a721a12bbc803b44f645561ecb1e1",
"/assets/packages/cupertino_icons/assets/CupertinoIcons.ttf": "9a62a954b81a1ad45a58b9bcea89b50b",
"/assets/AssetManifest.json": "cc5894ec3c840a4a6ca93b27b8e8178a",
"/assets/LICENSE": "80fd00b2a4010432664c5aaf8ac00704",
"/assets/FontManifest.json": "01700ba55b08a6141f33e168c4a6c22f",
"/assets/CHANGELOG.md": "b6d57b5714b9ce36fb30ce40a715b2ca",
"/assets/fonts/MaterialIcons-Regular.ttf": "56d3ffdef7a25659eab6a68a3fbfaf16",
"/main.dart.js": "bd91b849053cb9a5d23c43c60cda923e",
"/favicon.png": "5dcef449791fa27946b3d35ad8803796"
};

self.addEventListener('activate', function (event) {
  event.waitUntil(
    caches.keys().then(function (cacheName) {
      return caches.delete(cacheName);
    }).then(function (_) {
      return caches.open(CACHE_NAME);
    }).then(function (cache) {
      return cache.addAll(Object.keys(RESOURCES));
    })
  );
});

self.addEventListener('fetch', function (event) {
  event.respondWith(
    caches.match(event.request)
      .then(function (response) {
        if (response) {
          return response;
        }
        return fetch(event.request, {
          credentials: 'include'
        });
      })
  );
});


lib/flutter_whatsnew.dart
library flutter_whatsnew;

import 'dart:io';

import 'package:flutter/cupertino.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_markdown/flutter_markdown.dart';

class WhatsNewPage extends StatelessWidget {
  final Widget title;
  final Widget buttonText;
  final List<ListTile> items;
  final VoidCallback onButtonPressed;
  final bool changelog;
  final String changes;
  final Color backgroundColor;
  final Color buttonColor;
  final String path;

  const WhatsNewPage({
    @required this.items,
    @required this.title,
    @required this.buttonText,
    this.onButtonPressed,
    this.backgroundColor,
    this.buttonColor,
  })  : changelog = false,
        changes = null,
        path = null;

  const WhatsNewPage.changelog({
    @required this.title,
    @required this.buttonText,
    this.onButtonPressed,
    this.changes,
    this.backgroundColor,
    this.buttonColor,
    this.path,
  })  : changelog = true,
        items = null;

  static void showDetailPopUp(
      BuildContext context, String title, String detail) async {
    void showDemoDialog<T>({BuildContext context, Widget child}) {
      showDialog<T>(
        context: context,
        barrierDismissible: false,
        builder: (BuildContext context) => child,
      );
    }

    return showDemoDialog<Null>(
      context: context,
      child: AlertDialog(
        title: Text(title),
        content: Text(detail),
        actions: <Widget>[
          FlatButton(
            child: Text('OK'),
            onPressed: () {
              Navigator.pop(context);
            },
          ),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    print("Changelog: $changelog");
    if (!kIsWeb && Platform.isIOS) {
      return _buildIOS(context);
    }

    return _buildAndroid(context);
  }

  Widget _buildAndroid(BuildContext context) {
    if (changelog) {
      return Scaffold(
        backgroundColor:
            backgroundColor ?? Theme.of(context).scaffoldBackgroundColor,
        body: SafeArea(
          child: Stack(
            fit: StackFit.loose,
            children: <Widget>[
              Positioned(
                top: 10.0,
                left: 0.0,
                right: 0.0,
                child: title,
              ),
              Positioned(
                left: 0.0,
                right: 0.0,
                top: 50.0,
                bottom: 80.0,
                child: ChangeLogView(changes: changes, path: path),
              ),
              Positioned(
                bottom: 5.0,
                right: 10.0,
                left: 10.0,
                child: ListTile(
                  title: RaisedButton(
                    child: buttonText,
                    color: buttonColor ?? Colors.blue,
                    onPressed: onButtonPressed != null
                        ? onButtonPressed
                        : () {
                            Navigator.pop(context);
                          },
                  ),
                ),
              ),
            ],
          ),
        ),
      );
    }
    return Scaffold(
      backgroundColor:
          backgroundColor ?? Theme.of(context).scaffoldBackgroundColor,
      body: SafeArea(
        child: Stack(
          fit: StackFit.loose,
          children: <Widget>[
            Positioned(
              top: 10.0,
              left: 0.0,
              right: 0.0,
              child: title,
            ),
            Positioned(
              left: 0.0,
              right: 0.0,
              top: 50.0,
              bottom: 80.0,
              child: ListView(
                children: items
                    .map(
                      (ListTile item) => ListTile(
                        title: item.title,
                        subtitle: item.subtitle,
                        leading: item.leading,
                        trailing: item.trailing,
                        onTap: item.onTap,
                        onLongPress: item.onLongPress,
                      ),
                    )
                    .toList(),
              ),
            ),
            Positioned(
              bottom: 5.0,
              right: 10.0,
              left: 10.0,
              child: ListTile(
                title: RaisedButton(
                  child: buttonText,
                  color: buttonColor ?? Colors.blue,
                  onPressed: onButtonPressed != null
                      ? onButtonPressed
                      : () => Navigator.pop(context),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildIOS(BuildContext context) {
    Widget child;
    if (changelog) {
      child = ChangeLogView(changes: changes, path: path);
    } else {
      child = Material(
        child: ListView(
          children: items,
        ),
      );
    }
    return CupertinoPageScaffold(
      navigationBar: CupertinoNavigationBar(
        middle: title,
        automaticallyImplyMiddle: false,
        trailing: CupertinoButton(
          child: buttonText,
          onPressed: onButtonPressed != null
              ? onButtonPressed
              : () {
                  Navigator.pop(context);
                },
        ),
      ),
      child: SafeArea(
        child: child,
      ),
    );
  }
}

class ChangeLogView extends StatefulWidget {
  const ChangeLogView({this.changes, this.path});
  final String changes;
  final String path;
  @override
  _ChangeLogViewState createState() => _ChangeLogViewState();
}

class _ChangeLogViewState extends State<ChangeLogView> {
  String _changelog;

  @override
  void initState() {
    if (widget?.changes == null) {
      rootBundle.loadString(widget?.path ?? "CHANGELOG.md").then((data) {
        setState(() {
          _changelog = data;
        });
      });
    } else {
      setState(() {
        _changelog = widget.changes;
      });
    }
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    if (_changelog == null) {
      return CircularProgressIndicator();
    }
    return Markdown(data: _changelog);
  }
}


lib/example/example.dart
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';
import 'dart:io';
import 'dart:async';
import 'package:native_widgets/native_widgets.dart';
import 'package:flutter_whatsnew/flutter_whatsnew.dart';

class ShowWhatsNew extends StatefulWidget {
  ShowWhatsNew({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _ShowWhatsNewState createState() => _ShowWhatsNewState();
}

class _ShowWhatsNewState extends State<ShowWhatsNew> {
  
  final double textScaleFactor = 1.0;

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      body: SafeArea(
        //Whats New Page is Just A Widget, Use in a Scaffold and SafeArea for Best Results
        child: WhatsNewPage(
          title: Text(
            "What's New",
            textScaleFactor: textScaleFactor,
            textAlign: TextAlign.center,
            style: TextStyle(
              // Text Style Needed to Look like iOS 11
              fontSize: 22.0,
              fontWeight: FontWeight.bold,
            ),
          ),
          buttonText: Text(
            'Continue',
            textScaleFactor: textScaleFactor,
            style: TextStyle(
              color: Colors.white,
            ),
          ),
          // Create a List of WhatsNewItem for use in the Whats New Page
          // Create as many as you need, it will be scrollable
          items: <ListTile>[
            ListTile(
              leading: Icon(Icons.color_lens),
              title: Text(
                'Dark Theme',
                textScaleFactor: textScaleFactor,
              ), //Title is the only Required Item
              subtitle: Text(
                'Black and grey theme (Tap to Change)',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                // You Can Navigate to Locations in the App
                Navigator.of(context).pushNamed("/settings");
              },
            ),
            ListTile(
              leading: Icon(Icons.map),
              title: Text(
                'Google Maps',
                textScaleFactor: textScaleFactor,
              ),
              subtitle: Text(
                'Open Address Links in Google Maps instead of Apple Maps (Tap to Change)',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                // You Can Navigate to Locations in the App
                Navigator.of(context).pushNamed("/settings");
              },
            ),
            ListTile(
              leading: Icon(Icons.notifications_active),
              title: Text(
                'Push Notifications',
                textScaleFactor: textScaleFactor,
              ),
              subtitle: Text(
                'Stay tuned for important information that can be pushed to you',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                WhatsNewPage.showDetailPopUp(
                  context,
                  'Info',
                  "You can turn off push notifications any time in your application settings.",
                );
              },
            ),
            ListTile(
              leading: Icon(Icons.arrow_drop_down),
              title: Text(
                'Dropdown Actions',
                textScaleFactor: textScaleFactor,
              ),
              subtitle: Text(
                'New ability to call cell, home and office number from the list dropdown',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                // You Can Navigate to Locations in the App
                 WhatsNewPage.showDetailPopUp(
                  context,
                  'Info',
                  "Tap on the down arrow on any list item to display phone numbers and emails.",
                );
              },
            ),
            ListTile(
              leading: Icon(Icons.info),
              title: Text(
                'Dashboard Alert Icons',
                textScaleFactor: textScaleFactor,
              ),
              subtitle: Text(
                'Depending on the alert type, the icon and color for the list item will change',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                // You Can Navigate to Locations in the App
                Navigator.of(context).pushNamed("/menu");
              },
            ),
            ListTile(
              leading: Icon(Icons.notifications_none),
              title: Text(
                'Loan Milestone Icons',
                textScaleFactor: textScaleFactor,
              ),
              subtitle: Text(
                'Depending on the Loan Milestone, the icon and color for the list item will change',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                // You Can Navigate to Locations in the App
                Navigator.of(context).pushNamed("/menu");
              },
            ),
            ListTile(
              leading: Icon(Icons.person_outline),
              title: Text(
                'Loan Contacts Enhancements',
                textScaleFactor: textScaleFactor,
              ),
              subtitle: Text(
                'Updated look for faster navigation',
                textScaleFactor: textScaleFactor,
              ),
              onTap: () {
                 WhatsNewPage.showDetailPopUp(
                  context,
                  'Info',
                  "Navigate to any loan then select the bottom right icon to go to the contacts. You can press the dropdown arrow for contact information.",
                );
              },
            ),
          ], //Required
          home: HomePage(),
          showNow: false,
          showOnVersionChange: true,
        ),
      ),
    );
  }
}

// Stateful widget for managing name data
class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

// State for managing fetching name data over HTTPP>
class _HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        // backgroundColor: globals.isDarkTheme ? null : Colors.white,
        title: Text(
          "Home",
          // textScaleFactor: globals.textScaleFactor,
        ),
      ),
      body: SingleChildScrollView(
          child: SafeArea(
        child: ListBody(
          children: <Widget>[
            Container(
              height: 10.0,
            ),
          ],
        ),
      )),
    );
  }
}
